File: c,c++/panstack.cpp
#include<iostream>
#include<cstdio>
#include<algorithm>
#include<math.h>
#define MOD 1000000007
using namespace std;

int main()
{
	int test,n,i;
	scanf("%d",&test);
	
	while(test--)
	{
		long long int p=1;
		
		scanf("%d",&n);
		
		for(i=0;i<n-1;++i)
		{
			p*=n;
			if(p>=MOD)
				p=p%MOD;
		}
		printf("%lld\n",p);
	}
	
	return 0;
}

File: py/panstack.py
MOD=1000000007
	
#test=int(raw_input())
test=1
while test>0:
	for n in range(600,1001):
		initial=1;
		for i in range(1,n):
			initial=3*initial-1;
		initial=initial%MOD
		print "%d," % initial,
		test=test-1
